Sources/CoreMetrics/SystemMetrics.swift

```diff
@@ -0,0 +1,98 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Metrics API open source project
+//
+// Copyright (c) 2018-2019 Apple Inc. and the Swift Metrics API project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of Swift Metrics API project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+
+public enum SystemMetricsProvider {
+    fileprivate static let lock = ReadWriteLock()
+    fileprivate static let queue = DispatchQueue(label: "com.apple.CoreMetrics.SystemMetricsProvider")
+    fileprivate static var shouldRunSystemMetrics: Bool = false
+
+    public static func bootstrapSystemMetrics() {
```

Comment:
would making this an extension of `MetricsSystem` make sense? something like 

`MetricsSystem. monitorSystem()` or  `MetricsSystem.startSystemMonitor()`

Reply:
Reason for this is because I can not access the lock. However if we don't use the `shouldRunSystemMetrics` anymore but use a `Cancellable` like you suggested, that issue would be resolved. Will look into this.

Reply:
Addressed all other comments, and seems like we'll still ned a Lock of some sorts. Since the one in `MetricsSystem` is `fileprivate` this will have to be a separate object (or we make the lock non-fileprivate)